pkg/apis/monitoring/v1/types.go

```diff
@@ -288,6 +288,16 @@ type WebConfigFileFields struct {
 	TLSConfig *WebTLSConfig `json:"tlsConfig,omitempty"`
 	// Defines HTTP parameters for web server.
 	HTTPConfig *WebHTTPConfig `json:"httpConfig,omitempty"`
+	// Defines the basic auth users for server.
+	// +optional
+	BasicAuthUsers []*WebBasicAuthUsers `json:"basicAuthUsers,omitempty"`
+}
+
+// WebBasicAuthUsers defines the basic auth users for --web.config.file flag.
+// +k8s:openapi-gen=true
+type WebBasicAuthUsers struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
```

Comment:
you don't want passwords in clear text inside the Prometheus CRD.

Reply:
yes, i'm on the way to slove this. i'm going to store `username` and `password` in secret, and when reconcile `Prometheus` resource, we can get info from secret and then set basic auth. Also, set probe to solve the kubernetes readiness check.

Reply:
It's not acceptable from a security standpoint to store credentials in clear-text in the statefulset spec. We need https://github.com/prometheus/exporter-toolkit/pull/151 to allow unauthenticated access to the probe endpoints.